views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,532 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Test Previewer Responsive Scale plugin : Scale
+ *
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'ui/hider',
+    'taoTests/runner/plugin',
+    'taoQtiTestPreviewer/previewer/utils/devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/preview-types',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/mobile-devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/desktop-devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/scale-wrapper',
+    'ui/selecter',
+    'css!taoItemsCss/preview'
+], function (
+    $,
+    _,
+    __,
+    hider,
+    pluginFactory,
+    devices,
+    previewTypesTpl,
+    mobileDevicesTpl,
+    desktopDevicesTpl,
+    scaleWrapperTpl,
+    selecter
+) {
+    'use strict';
+
+    var orientation = 'landscape';
+    var DEFAULT_TYPE = 'standard';
+    var selectedDevice = 'standard';
+
+    var deviceTypes = {
+            desktop: __('Desktop preview'),
+            mobile: __('Mobile preview'),
+            standard: __('Actual size')
+        };
+
+    var $window = $(window);
+
+    var screenSize = {
+        width: $window.innerWidth(),
+        height: $window.innerHeight()
+    };
+
+    var sizeSettings = {
+        width: 0,
+        height: 0
+    };
+
+    var devicesToControls = {
+        'mobile': '$mobileDevices',
+        'desktop': '$desktopDevices',
+        'standard': '$deviceTypes'
+    };
+
+    /**
+     * prepares data for list of devices user would chose from.
+     * @returns {Array} - list of devices user would work throug with help of ui-controls
+     * @private
+     */
+    var _getPreviewTypes = function () {
+        var options = [];
+        _(deviceTypes).forEach(function (_deviceLabel, _deviceType) {
+            options.push({
+                value: _deviceType,
+                label: _deviceLabel,
+                selected: selectedDevice === _deviceType
+            });
+        });
+        return options;
+    };
+
+    var controls;
+
+    var testRunner;
+
+    var api = {
+        /**
+         * Tells if the component is enabled
+         * @returns {Boolean}
+         */
+        isPluginAllowed: function isPluginAllowed() {
+            var config = testRunner.getConfig();
```

Comment:
Should be `this.getTestRunner()`.
But, anyway. This method won't be reachable by the plugin.